staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -955,8 +981,13 @@ func (o *ApplyOptions) PrintAndPrunePostProcessor() func() error {
 		}
 
 		if o.Prune {
-			p := newPruner(o)
-			return p.pruneAll(o)
+			if cmdutil.AlphaEnabled(cmdutil.ApplySetEnv) && o.ApplySet != nil {
```

Comment:
Nit: if we don't already have a feature flag type and we wanted to be a bit more type-safe, we could do something like:

```
type FeatureFlag string
var FeatureFlag ApplySetEnv = "FOO"

func (f *FeatureFlag) IsEnabled() {
...
```

Reply:
Love this suggestion 👌